### DISCUSSION TITLE
protected $connection = 'custom_database' ignored

### DISCUSSION THREAD

**Discussion by elreco** [spatie/laravel-medialibrary#2238] [Q&A]
Hello,

We got an issue using laravel-medialibrary with two databases configured in our Laravel app.

We got two models :

**App\Models\User.php** using connection named 'mysql'

```php
<?php

namespace  App\Models;

use Spatie\MediaLibrary\HasMedia;
use Spatie\MediaLibrary\InteractsWithMedia;

use Illuminate\Database\Eloquent\Model;

class User extends Model  implements HasMedia
{
    use InteractsWithMedia;
    protected $connection = 'mysql'
}
```

and

**App\Models\Classe.php** using connection named 'custom_database'

```php
<?php

namespace  App\Models;

use Spatie\MediaLibrary\HasMedia;
use Spatie\MediaLibrary\InteractsWithMedia;

use Illuminate\Database\Eloquent\Model;

class Classe extends Model  implements HasMedia
{
    use InteractsWithMedia;
    protected $connection = 'custom_database'
}
```

Then I try to upload a file in my ClasseController : 

```php
 $classe = Classe::find($request->id);
 return $classe->addMedia($request->file('file'))->toMediaCollection('academy', 'academy')->getUrl();
```

Laravel-medialibrary is always saving images in 'mysql' database :/

We tried to create a custom Media class but it doesn't work.

Do you have a solution?

Thank you!

**Comment by freekmurze** [comment]
I'd welcome a PR that adds a test to demonstrate this bug and a PR that fixes this.
